Unfortunately build on `aarch64-darwin` is failing on `tiledb` (due to update in #386586)

```
[  0%] Building CXX object tiledb/sm/misc/CMakeFiles/math.dir/tdb_math.cc.o
CMake Error: Unable to read from file '../../../../../../../nix/store/6dhmi0kl8i6vi3m1j6w23g70c2vhyj5b-file-5.46/share/misc/magic.mgc': Can't lstat ../../../../../../../nix/store/6dhmi0kl8i6vi3m1j6w23g70c2vhyj5b-file-5.46/share/misc/magic.mgc
CMake Error at /tmp/nix-build-tiledb-2.27.2.drv-0/source/tiledb/sm/misc/generate_embedded_data_header.script.cmake:43 (file):
  file failed to compress:
  /tmp/nix-build-tiledb-2.27.2.drv-0/source/build/magic.mgc.zst
```

--- a/pkgs/development/libraries/mapnik/default.nix
+++ b/pkgs/development/libraries/mapnik/default.nix
@@ -29,13 +29,13 @@
 
 stdenv.mkDerivation rec {
   pname = "mapnik";
-  version = "4.0.5";
+  version = "4.0.7";
 
   src = fetchFromGitHub {
     owner = "mapnik";
     repo = "mapnik";
     rev = "v${version}";
-    hash = "sha256-pReoyMdu8RYrberKcwGw0DKmkxVRJezZYcPAM/UAn6o=";
+    hash = "sha256-gJktRWcJiSGxxjvWFt+Kl9d7g+TOSPk2PfGP0LIVxt4=";
     fetchSubmodules = true;
   };
 
@@ -43,7 +43,7 @@ stdenv.mkDerivation rec {
 
   postPatch = ''
     substituteInPlace configure \
-      --replace '$PYTHON scons/scons.py' ${buildPackages.scons}/bin/scons
+      --replace-fail '$PYTHON scons/scons.py' ${buildPackages.scons}/bin/scons
     rm -r scons
     # Remove bundled 'sparsehash' directory in favor of 'sparsehash' package
     rm -r deps/mapnik/sparsehash
@@ -134,15 +134,5 @@ stdenv.mkDerivation rec {
       ];
     license = licenses.lgpl21Plus;
     platforms = platforms.all;
-
-    # 29-03-2025: On darwin, the libc++ standard library is used to compile C++ programs.
-    # 29-03-2025: Since the base template for `std::char_traits` was removed in LLVM 19,
-    # 29-03-2025: usages of `boost::u32regex` will no longer compile.
-    # 29-03-2025: Linux builds do not fail as they use libstdc++, which has not removed
-    # 29-03-2025: such `std::char_trait` declarations.
-    #
-    # 29-03-2025: See https://github.com/mapnik/mapnik/issues/4499 for more information
-    # 29-03-2025: and a Minimal Reproducible Example.
-    badPlatforms = platforms.darwin;
   };
 }

